pkg/webhook/pod/pods.go

```diff
@@ -239,10 +259,34 @@ func (pc *PodAdmissionControl) AdmitCreatePods(ar admission.AdmissionReview) *ad
 		return util.ARFail(err)
 	}
 
+	if memberUtils.NeedForceUpgrade(tc) {
+		klog.Infof("tc[%s/%s] is force upgraded, admit to create pod[%s/%s]", namespace, tcName, namespace, name)
+		return util.ARSuccess()
+	}
+
 	if l.IsTiKV() {
 		pdClient := pc.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.Spec.EnableTLSCluster)
 		return pc.admitCreateTiKVPod(pod, tc, pdClient)
 	}
 
 	return util.ARSuccess()
 }
+
+func fetchInfoFromPayload(payload *admitPayload) (pod *core.Pod, ownerStatefulSet *apps.StatefulSet, ordinal int32, podName, namespace, tcName string, isInOrdinal, isUpgrading, IsDeferDeleting bool, pdClient pdapi.PDClient, err error) {
```

Comment:
Why so many return values? For the values that can be referred to with payload directly, I think no need to return them in another function.
@weekface

Reply:
updated.